src/redux/missions/missions.js

```diff
@@ -36,10 +40,24 @@ export const fetchMissionsFromApi = () => async (dispatch) => {
 // reducers
 const reducer = (state = initialState, action) => {
   switch (action.type) {
-    case JOIN_MISSION:
-      return [...state, action.payload];
-    case LEAVE_MISSION:
-      return state.filter((mission) => mission.id !== action.payload);
+    case JOIN_MISSION: {
+      const newState = state.map((mission) => {
+        if (mission.mission_id !== action.payload.id) {
+          return mission;
+        }
+        return { ...mission, reserved: true };
+      });
+      return [...newState];
+    }
+    case LEAVE_MISSION: {
+      const newState = state.map((mission) => {
+        if (mission.mission_id !== action.payload.id) {
+          return mission;
+        }
+        return { ...mission, reserved: false };
+      });
+      return [...newState];
+    }
```

Comment:
Great getting the ID of the reserved Mission and updating the state. Returned a new state object with `Joined to Mission`